providers/mongo/shedlock-provider-mongo/src/main/java/net/javacrumbs/shedlock/provider/mongo/MongoLockProvider.java

```diff
@@ -146,11 +143,22 @@ public boolean updateRecord(LockConfiguration lockConfiguration) {
                 set(LOCKED_AT, now),
                 set(LOCKED_BY, hostname)
             );
-            Document result = getCollection().findOneAndUpdate(
-                and(eq(ID, lockConfiguration.getName()), lte(LOCK_UNTIL, now)),
-                update
-            );
-            return result != null;
+            try {
+                Document result = getCollection().findOneAndUpdate(
+                        and(eq(ID, lockConfiguration.getName()), lte(LOCK_UNTIL, now)),
+                        update,
+                        new FindOneAndUpdateOptions().upsert(true).returnDocument(ReturnDocument.AFTER)
+                );
+                return result != null;
```

Comment:
Is the result ever null?

Reply:
According to MongoDB docs it should not, but I like my contracts in types, not words. For the current state of Mongo driver this could just be `true`.